packages/core/src/lib/filter/index.ts

```diff
@@ -72,7 +73,7 @@ class Subscription {
   async subscribe<T extends IDecodedMessage>(
     decoders: IDecoder<T> | IDecoder<T>[],
     callback: Callback<T>
-  ): Promise<void> {
+  ): Promise<IFilterResponse> {
```

Comment:
This function can fail in several manners:
1. locally (can't encode, other checks fail)
2. on the communication (can't open stream, remote node does not reply)
3. Remote node send an error back


All these cases should be covered with an Enum

Only in the case of (3) then you may want to return a "sub error", ie, what the remote peer returned.

Currently, this is still throwing exception on (1) and (2). It should not.

Finally, there is an opportunity to have common errors with light push. `SendError` should be re-used here and made more generic.

Do not we will change light push to also return error like filter, so while it's not there yet, knowing that should allow a common design. E.g. today light push remote peer failure does not include sub error, tomorrow it will.